src/object/srv_obj.c

```diff
@@ -955,11 +955,13 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 
 	if (rma) {
 		bulk_bind = orw->orw_flags & ORF_BULK_BIND;
-		if (oca->ca_resil == DAOS_RES_EC) {
-			rc = obj_ec_update_bulk_transfer(rpc, bulk_bind,
-						     orw->orw_bulks.ca_arrays,
-						     ioh, skip_list,
-						     orw->orw_nr);
+		if ((opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_UPDATE ||
+		     opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_TGT_UPDATE ||
+		     opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_FETCH) &&
+			oca->ca_resil == DAOS_RES_EC) {
```

Comment:
Minor, this should probably be checked first so && will short_circuit the other 3 checks if it's not EC